tracer/src/Datadog.Trace/Iast/StackWalker.cs

```diff
@@ -66,12 +51,50 @@ public static StackFrameInfo GetFrame()
             }
 
             var assembly = declaringType?.Assembly.GetName().Name;
-            if (assembly != null && !AssemblyNamesToSkip.Contains(assembly))
+            if (assembly != null && !AssemblyExcluded(assembly))
             {
                 return new StackFrameInfo(frame, true);
             }
         }
 
         return new StackFrameInfo(null, true);
     }
+
+    public static bool AssemblyExcluded(string assembly)
+    {
+        if (ExcludedAssemblyCache.TryGetValue(assembly, out bool excluded))
+        {
+            return excluded;
+        }
+
+        excluded = IsExcluded(assembly);
+        ExcludedAssemblyCache[assembly] = excluded;
+
+        return excluded;
+    }
+
+    // For performance reasons, we are not supporting wildcards fully. We just need to use '*' at the end for now. We can use regular expressions
+    // if in the future we need a more sophisticated wildcard support
+    private static bool IsExcluded(string assembly)
+    {
```

Comment:
I'd rather use case insensitive comparisons

Reply:
For me, both options are fine: while we should add the correct assembly names, including casing, I guess that there might be some specific cases where some version of the assembly name could have a different casing... Anyway, I have modified it.